File: src/linkerz_crawler_core/src/main/java/org/linkerz/crawler/core/downloader/Downloader.java
/*
 * Copyright (C) 2012 - 2013 LinkerZ
 */

package org.linkerz.crawler.core.downloader;

import org.linkerz.crawler.core.downloader.result.DownloadResult;
import org.linkerz.crawler.core.model.WebLink;

import java.io.Serializable;

/**
 * The Class Downloader.
 *
 * @author Nguyen Duc Dung
 * @since 7/2/12, 12:25 AM
 */
public interface Downloader<R extends DownloadResult> extends Serializable {
    public DownloadResult download(WebLink webLink) throws Exception;
}

File: src/linkerz_crawler_core/src/main/scala/org/linkerz/crawler/core/downloader/Downloader.scala
/*
 * Copyright (C) 2012 - 2013 LinkerZ
 */

package org.linkerz.crawler.core.downloader

import org.linkerz.crawler.core.model.WebUrl
import com.ning.http.client.{AsyncHttpClient, AsyncHttpClientConfig}
import grizzled.slf4j.Logging

/**
 * The Class Downloader.
 *
 * @author Nguyen Duc Dung
 * @since 7/29/12, 1:07 AM
 *
 */

trait Downloader extends Logging {

  def download(webUrl: WebUrl): DownloadResult

}
